tests/test_flow_studio_ui_ids.py

```
@@ -93,12 +93,15 @@ def extract_uiids_from_html(html: str) -> List[Tuple[str, int]]:
     for line_num, line in enumerate(html.split("\n"), start=1):
         # Handle script tag transitions
         if script_start.search(line):
-            in_script = True
+            # Do NOT treat data-inline-source as a regular script block to skip
+            if "data-inline-source=" not in line:
+                in_script = True
         if script_end.search(line):
-            in_script = False
-            continue  # Skip the closing script line
+            if in_script:
```

Ruff (0.15.5)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)